pex/interpreter.py

```diff
@@ -400,11 +411,20 @@ def _execute(cls, binary, args=None, pythonpath=None, env=None, stdin_payload=No
 
     @classmethod
     def _spawn_from_binary_external(cls, binary):
-        def create_interpreter(stdout):
+        def create_interpreter(stdout, check_binary=False):
             identity = stdout.decode("utf-8").strip()
             if not identity:
-                raise cls.IdentificationError("Could not establish identity of %s" % binary)
-            return cls(PythonIdentity.decode(identity))
+                raise cls.IdentificationError("Could not establish identity of {}.".format(binary))
+            interpreter = cls(PythonIdentity.decode(identity))
+            # We should not need to check this since binary == interpreter.binary should always be
+            # true, but historically this could be untrue as noted in `PythonIdentity.get`.
+            if check_binary and not os.path.exists(interpreter.binary):
+                raise cls.InterpreterNotFound(
+                    "Cached interpreter for {} reports a binary of {} which is not found".format(
```

Comment:
```suggestion
                    "Cached interpreter for {} reports a binary of {}, which could not be found".format(
```

Reply:
Is there any remedy here? Could we instruct the user to purge the cache, for example?

Reply:
The code you commented on above does purge the cache automatically already. This log line at finest log level is just to say FYI for someone debugging things deeply.

Reply:
I'm not following that. This raises an exception, rather than logging. Where is the cache purge happening?